lesson34/src/js/article.js

```diff
@@ -0,0 +1,159 @@
+const articleWrapper = document.getElementById("js-article-wrapper");
+let currentArticleData;
+
+const createElementWithClassName = (type, className) => {
+  const element = document.createElement(type);
+  element.className = className;
+  return element;
+};
+
+const createLoading = () => {
+  const loadingWrapper = createElementWithClassName("div", "loading-wrapper");
+  const loading = document.createElement("img");
+  loadingWrapper.id = "js-loading";
+  loading.src = "./img/loading-circle.gif";
+  loadingWrapper.appendChild(loading);
+  return loadingWrapper;
+};
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const createErrorMessage = (error) => {
+  const errorMessage = createElementWithClassName("p", "article-error-message");
+  errorMessage.textContent = error;
+  return errorMessage;
+};
+
+const fetchErrorHandling = async (response) => {
+  if (response.ok) {
+    return await response.json();
+  } else {
+    const responseMessage = `${response.status}:${response.statusText}`;
+    articleWrapper.appendChild(createErrorMessage(responseMessage));
+    console.error(responseMessage);
+  }
+};
+
+const getJsonOrError = async (url) => {
+  const response = await fetch(url);
+  const json = await fetchErrorHandling(response);
+  return json;
+};
+
+const getArticleData = async (url) => {
+  articleWrapper.appendChild(createLoading());
+  try {
+    return await getJsonOrError(url)
+  } catch (error) {
+    articleWrapper.appendChild(createErrorMessage(`データの読み取りに失敗しました${error}`));
+  } finally {
+    removeLoading();
+  }
+}
+
+const registerFavoriteButton = () => {
+  const favoriteButton = document.getElementById("js-favorite-button");
+  favoriteButton.classList.add("is-favorite");
+  favoriteButton.dataset.favoriteAction = "registered";
+  favoriteButton.firstChild.src = "./img/star-icon--disabled.svg";
+  favoriteButton.disabled = true;
+}
+
+const setFavoriteArticleDataInLocalStorage = () => {
+  const loginUserToken = localStorage.getItem("token");
+  const favoriteArticlesData = JSON.parse(localStorage.getItem("favoriteArticles")) ?? {};
+  const favoriteArticlesDataOfLoginUser = favoriteArticlesData[loginUserToken] ?? [];
+  const registeredFavoriteArticleData = [
+    ...favoriteArticlesDataOfLoginUser,
+    {
+      [currentArticleData.id]: {
+        "title": currentArticleData.title,
+        "date": currentArticleData.date,
+        "thumbnail": currentArticleData.thumbnail
+      }
+    }
+  ]
+  const margeData = { ...favoriteArticlesData, [loginUserToken]: registeredFavoriteArticleData };
```

Comment:
```suggestion
  const margedData = { ...favoriteArticlesData, [loginUserToken]: registeredFavoriteArticleData };
```

Reply:
Thank you.I changed it to past tense because it stores the data after the merge.
[d1255f0](https://github.com/sae-github/JS_lesson/pull/42/commits/d1255f0cef510f2cb0d5ce036117d4e9f0a10091)